src/ORM/Table.php

```diff
@@ -389,10 +389,10 @@ public function registryAlias($registryAlias = null)
     /**
      * Returns the connection instance or sets a new one
      *
-     * @param \Cake\Database\Connection|null $conn The new connection instance
-     * @return \Cake\Database\Connection
+     * @param \Cake\Datasource\ConnectionInterface|null $conn The new connection instance
+     * @return \Cake\Datasource\ConnectionInterface
      */
-    public function connection(Connection $conn = null)
+    public function connection(ConnectionInterface $conn = null)
```

Comment:
We would need to communicate that as it will break userland typehints in overwriting/extending classes, will it not?